### PR TITLE
bug(emp-creator): add financialProductLib address to deployment params

### LINKED ISSUE #2233: Add financial product library address parameter to ExpiringMultiPartyCreator

### PR DESCRIPTION
**Pull request by nicholaspai**
Signed-off-by: Nick Pai <npai.nyc@gmail.com>

**Motivation**

#2233 

**Details**

Allows deployer to specify a FPLib when deploying a new EMP. Previous behavior would default all EMP's to not using a FPLib, which isn't a major bug but doesn't take advantage of the FPLib contract param.

**Issue(s)**

Fixes #2233

### CODE REVIEW

File: packages/core/test/financial-templates/expiring-multiparty/ExpiringMultiPartyCreator.js [hunk 1]
```diff
@@ -293,4 +294,27 @@ contract("ExpiringMultiPartyCreator", function(accounts) {
...
+      return ev.expiringMultiPartyAddress != 0 && ev.deployerAddress == contractCreator;
+    });
+    let expiringMultiParty = await ExpiringMultiParty.at(expiringMultiPartyAddress);
+    let linkedFPLib = await StructuredNoteFinancialProductLibrary.at(
+      await expiringMultiParty.financialProductLibrary()
+    );
+
+    // FPLib address is saved correctly.
+    assert.equal(linkedFPLib.address, structuredNoteFPL.address);
+  });
```
Comment (nicholaspai) [inline packages/core/test/financial-templates/expiring-multiparty/ExpiringMultiPartyCreator.js:319]: @UMAprotocol/eng Should any other tests be done here to test that the EMP is "correct linked" to the FPLib?

**Review by chrismaree — APPROVED** [review]
Looks good! Thanks for fixing this

File: packages/core/test/financial-templates/expiring-multiparty/ExpiringMultiPartyCreator.js (hunk 1 above)
Comment (chrismaree) [reply packages/core/test/financial-templates/expiring-multiparty/ExpiringMultiPartyCreator.js:319]: Can try calling a transformation on a deployed emp that uses the transformation such as one of the public view functions? Could also check that the deployed emp has the library set correctly to the address you created

Comment (chrismaree) [reply packages/core/test/financial-templates/expiring-multiparty/ExpiringMultiPartyCreator.js:319]: You've done the second test actually already

**Review by daywiss — APPROVED** [review]

File: packages/core/test/financial-templates/expiring-multiparty/ExpiringMultiPartyCreator.js (hunk 1 above)
Comment (mrice32) [reply packages/core/test/financial-templates/expiring-multiparty/ExpiringMultiPartyCreator.js:319]: IMO, the unit tests for the emp should cover whether the address returned by the public financialProductLibrary() method match the transformations done in the contract, and that sort of test shouldn't be necessary for the scope of this contract. The creator tests should just be sure that the address is injected properly.

But more tests wouldn't necessarily hurt :).

**Review by mrice32 — APPROVED** [review]
LGTM! Thanks for adding this!

**Comment by nicholaspai** [comment]
@chrismaree @mrice32 
> IMO, the unit tests for the emp should cover whether the address returned by the public financialProductLibrary() method match the transformations done in the contract, and that sort of test shouldn't be necessary for the scope of this contract. The creator tests should just be sure that the address is injected properly.

I  agree with this, because the `financial-product-library` tests should validate that if the FPLib is properly loaded, the transformations work. We don't want this creator test to be opinionated about what a correct "transformation" looks like.

File: packages/core/test/financial-templates/expiring-multiparty/ExpiringMultiPartyCreator.js (hunk 1 above)
Comment (nicholaspai) [reply packages/core/test/financial-templates/expiring-multiparty/ExpiringMultiPartyCreator.js:319]: In response to first suggestion see [here](https://github.com/UMAprotocol/protocol/pull/2256#issuecomment-737513983)!

**Pull request merged**